src/queueing_honey_badger/mod.rs

```diff
@@ -101,12 +101,11 @@ where
     }
 
     /// Creates a new Queueing Honey Badger instance with an empty buffer.
-    pub fn build<R>(self, rng: R) -> QueueingHoneyBadgerWithStep<T, N, Q>
+    pub fn build<R>(self, rng: R) -> Result<QueueingHoneyBadgerWithStep<T, N, Q>>
     where
         R: 'static + Rng + Send + Sync,
     {
         self.build_with_transactions(None, rng)
-            .expect("building without transactions cannot fail")
```

Comment:
If it's true that this cannot fail it would be better to leave this `expect` here. There's no point in having a runtime error for something that cannot occur at runtime (and is not a consequence of programmer error).

I'm guessing there's more to the story though.